libsolidity/formal/SMTEncoder.cpp

```diff
@@ -201,7 +201,11 @@ void SMTEncoder::visitFunctionOrModifier()
 		if (dynamic_cast<ContractDefinition const*>(refDecl))
 			visitFunctionOrModifier();
 		else if (auto modifierDef = dynamic_cast<ModifierDefinition const*>(refDecl))
-			inlineModifierInvocation(modifierInvocation.get(), modifierDef);
+		{
+			solAssert(*modifierInvocation->name()->annotation().requiredLookup == VirtualLookup::Virtual, "");
```

Comment:
Why is this true?

Reply:
Not sure actually. But from the other places in the code, I see that this is usually checked before a call to `resolveVirtual`. Also in `ContractCompiler::appendModifierOrFunctionCode` from where I got this idea.

Reply:
Ok. It seems to work for normal modifiers as well, so fine.